src/AzureAppConfigurationImpl.ts

```diff
@@ -60,25 +101,111 @@ export class AzureAppConfigurationImpl extends Map<string, unknown> implements A
             };
             if (this.#requestTracingEnabled) {
                 listOptions.requestOptions = {
-                    customHeaders: this.#customHeaders()
+                    customHeaders: this.#customHeaders(requestType)
                 }
             }
 
             const settings = this.#client.listConfigurationSettings(listOptions);
 
             for await (const setting of settings) {
                 if (setting.key) {
-                    const [key, value] = await this.#processAdapters(setting);
-                    const trimmedKey = this.#keyWithPrefixesTrimmed(key);
-                    keyValues.push([trimmedKey, value]);
+                    const keyValuePair = await this.#processKeyValues(setting);
+                    keyValues.push(keyValuePair);
+                }
+                // update etag of sentinels if refresh is enabled
+                if (this.#refreshEnabled) {
+                    const matchedSentinel = this.#sentinels.find(s => s.key === setting.key && s.label === setting.label);
+                    if (matchedSentinel) {
+                        matchedSentinel.etag = setting.etag;
+                    }
                 }
             }
         }
+        this.clear(); // clear existing key-values in case of configuration setting deletion
         for (const [k, v] of keyValues) {
             this.set(k, v);
         }
     }
 
+    /**
+     * Refresh the configuration store.
+     */
+    public async refresh(): Promise<void> {
+        if (!this.#refreshEnabled) {
+            return Promise.resolve();
+        }
+
+        // if still within refresh interval/backoff, return
+        if (!this.#refreshTimer.canRefresh()) {
+            return Promise.resolve();
+        }
+
+        // try refresh if any of watched settings is changed.
+        let needRefresh = false;
+        for (const sentinel of this.#sentinels) {
+            let response: GetConfigurationSettingResponse | undefined;
+            try {
+                response = await this.#client.getConfigurationSetting(sentinel, {

```

Comment:
If a sentinel key was not loaded initially, its etag will be null. Will that possibly throw an error when calling `getConfigurationSetting` with `onlyIfChanged` set to true?

Reply:
Tried and it works as expected. The client returns configuration setting with status code 200.